### ISSUE TITLE
[App]: Downloading csv from the web app generates incoherent column names

### ISSUE THREAD

**Issue by pablo2909** [wandb/wandb#6474]
### Current Behavior

I generated a plot in a report from grouping line plots from multiple runs. I download then the csv associated to this plot to work on it. The plot looks like this: 

<img width="1644" alt="image" src="https://github.com/wandb/wandb/assets/40873702/3c484f73-b078-49e1-9eff-85bf1b0487b6">

When I download the csv and extract the column names I get this:

```bash
Index(['Training step', 'loss.beta_on_loss: true - _step',
       'loss.beta_on_loss: true - _step__MIN',
       'loss.beta_on_loss: true - _step__MAX',
       'loss.beta_on_loss: true - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: true - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: true - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:45:31 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:43:43 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:41:11 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:39:29 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:34:26 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:32:29 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:31:11 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:29:51 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl_2023-10-19 10:25:19 - Variance Loss CVL over Variance Training Loss__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - _step',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - _step__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - _step__MAX',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - Variance Loss CVL over Variance Training Loss',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - Variance Loss CVL over Variance Training Loss__MIN',
       'loss.beta_on_loss: 2gaussians_dsm_cvl - Variance Loss CVL over Variance Training Loss__MAX'],
      dtype='object')
```
The naming is incoherent. I do have a parameter `loss.beta_on_loss` but I don't understand why some columns have `loss.beta_on_loss: true` and others nothing. Also in the column names I do recognise the names I gave to my runs and that are grouped, but then a lot of the runs are missing. 

### Expected Behavior

I expect the column names to be the one of the legends.

### Environment

OS:
Mac OSX
Browsers:
Safari & Firefox 
Version:
0.14.0

### Additional Context

Thank you for any help

**Comment by JoanaMarieL** [comment]
Hello @pablo2909 ,

I am glad to assist your concern, may I please request for the actual link of your project workspace please?

**Comment by pablo2909** [comment]
Hi, thank you for your help, please find it [here](https://wandb.ai/pablo2909/SimpleScoreBased?workspace=user-pablo2909)

**Comment by sydholl** [comment]
WandB Internal User commented: 
pablo2909 commented: 
Hi, thank you for your help, please find it [here](https://wandb.ai/pablo2909/SimpleScoreBased?workspace=user-pablo2909)

**Comment by JoanaMarieL** [comment]
Thank you @pablo2909 for sharing your project workspace, we'll check this out and get back to you.

**Comment by sydholl** [comment]
WandB Internal User commented: 
JoanaMarieL commented: 
Thank you @pablo2909 for sharing your project workspace, we'll check this out and get back to you.

**Comment by pablo2909** [comment]
Any news ? :)

**Comment by pablo2909** [comment]
@sydholl @JoanaMarieL Anything ? :)

**Comment by umakrishnaswamy** [comment]
hey @pablo2909 - still looking into this - I was able to reproduce the behavior you're seeing and I have a few follow up questions:
- are you performing this grouping within the runs sidebar? or within the chart itself? any details about the steps you're taking to perform this grouping would be super helpful 
- were you able to see grouped runs exported as a csv in the past? any details about this would help a ton! 

thank you for your patience!

**Comment by pablo2909** [comment]
-  Yes the grouping is happening within the runs sidebar, the only thing I touch in the chart is changing the legend name and specifying that the spread computed in the grouping should be a standard error
- It's my first time experimenting with csv on wandb so I have no prior experience

Thank you for looking into it

**Comment by umakrishnaswamy** [comment]
hey @pablo2909! few updates:

- we currently do not support exporting grouped run metrics (which is why you're seeing a bunch of column names, reflective of individual runs, not the actual values of the grouped metrics). i will make a feature request for this and let you know of any plans there are to work on this & update you with any work
- were you able to see more of your runs in the csv once you toggled the visibility settings in the runs table? (the eye icon next to the runs) or is the csv download not responsive to those changes?
- for now I recommend exporting the csv without run grouping, then performing the grouping and data manipulation using pandas or another library 

please let me know if you have any questions about this, and i will be sure to update this github thread if any progress arises on the feature request

**Comment by pablo2909** [comment]
- Thank you, that would be a really cool feature imo and I hope it can happen :) 
- Yes toggling the eye icon impacts the number of runs exported in the csv
- I'll do that !

Thanks again for the help, I'll leave the issue open then to follow up on this feature request

**Comment by dudarboh** [comment]
Dear all, I have encountered a similar (or the same) issue. So, I would like to ping this thread once again to bring attention to it.
I am trying to export CSV data from my line chart in my project workspace, which looks like this:
<img src="https://github.com/user-attachments/assets/4d86a1e8-5fb4-44f7-a3a4-0b842be0cc52" width="100%">
I have stripped all other runs and made only a single run visible for simplicity.
Then I click three dots -> Download -> CSV and get this CSV file below.

![Image](https://github.com/user-attachments/assets/7caafc6f-fab9-4c27-a643-e4148f44e027)

What I assume is happening is that the column `epoch` which is used for the x-axis is used in all AVG/MIN/MAX calculations as a separate entry. It also has a weird shift of +1, which I am not sure where it comes from.

It destroys all the meaning of the exported data.
As this issue is at least two years old is there an update on the status or an already existing solution to simply export *raw* data via the UI interface?

**Comment by ArtsiomWB** [comment]
Hey @dudarboh, it looks like this ticket was processed and put into backlogs by our eng team. 

Could you please send me a workspace of where you are experiencing this, so I can re-escalate this to our Product Managers?

**Comment by dudarboh** [comment]
Hi @ArtsiomWB, I think I found the culprit of the problem and it is in "Advanced Legend".
Let me know if this link works for you:
https://wandb.ai/calo5d/TEST_PROJECT?nw=nwuserfoxwise

The reproducible from scratch is also quite simple
### Reproducible
1. Run the script
```
#!/usr/bin/env python3
import wandb

run = wandb.init(project="TEST_PROJECT")

for epoch in range(50):
    loss = 42
    run.log({"epoch":epoch, "loss": loss})

run.finish()
```

2. Plot in the UI "loss" vs "epoch"

3. Change the advanced legend to anything

4. Three dots -> export CSV is broken

5. Restoring advanced legend by hand does not fix CSV

**Comment by ArtsiomWB** [comment]
This is beautiful, thank you for the repro!

I will send it directly to our engineering team and will let you know about any updates I get regarding this issue. 

Warmly,
Artsiom
